wp-cli/class-option-autoload.php

```diff
@@ -0,0 +1,226 @@
+
+		if ( $check_option === $option_autoload ) {
+			WP_CLI::error( 'Option not updated.' );
+		}
+
+		$alloptions_before = wp_cache_get( 'alloptions', 'options' );
+		wp_cache_delete( 'alloptions', 'options' );
+		wp_load_alloptions(); // populate cache
+		$alloptions_after = wp_cache_get( 'alloptions', 'options' );
+
+		$cache_success = false;
+		switch ( $check_option ) {
+			case 'no':
+				$cache_success = ( isset( $alloptions_before[ $option ] ) ) && ( ! isset( $alloptions_after[ $option ] ) );
+				break;
+			case 'yes':
+				$cache_success = ( ! isset( $alloptions_before[ $option ] ) ) && ( isset( $alloptions_after[ $option ] ) );
+				break;
+		}
+
+		WP_CLI::success( sprintf(
+			'Autoload changed.%s',
+			WP_CLI::colorize( $cache_success ? ' Cache flushed.' : ' %rCache flush failed.%n' )
+		) );
+
+	}
+
+	/**
+	 * Get autoload for option
+	 *
+	 * ## OPTIONS
+	 *
+	 * <option>
+	 * : Name of option
+	 *
+	 * [--format=<format>]
+	 * : Format to use for the output. One of table, csv or json.
+	 * ---
+	 * default: table
+	 * options:
+	 *   - table
+	 *   - json
+	 *   - csv
+	 *   - yaml
+	 *   - count
+	 * ---
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     # Get autoload value for 'home'
+	 *     $ wp option autoload get home
+	 *     yes
+	 *
+	 *     # Get autoload value for 'home' as json
+	 *     $ wp option autoload get home --format=json
+	 *     "yes"
+	 *
+	 */
+	public function get( $args, $assoc_args ) {
+
+		list( $option ) = $args;
+
+		global $wpdb;
+		$option_autoload = $wpdb->get_var( $wpdb->prepare( "SELECT autoload from {$wpdb->options} where option_name = %s", $option ) );
+
+		if ( is_null( $option_autoload ) ) {
+			WP_CLI::error( 'Option does not exist.' );
+		}
+
+		WP_CLI::print_value( $option_autoload, $assoc_args );
+
+	}
+
+	/**
+	 * List all autoloaded (or not) options
+	 *
+	 * ## OPTIONS
+	 *
+	 * [<yn>]
+	 * : yes or no
+	 * ---
+	 * default: yes
+	 * options:
+	 *   - yes
+	 *   - no
+	 * ---
+	 *
+	 * [--format=<format>]
+	 * : Format to use for the output. One of table, csv or json.
+	 * ---
+	 * default: table
+	 * options:
+	 *   - table
+	 *   - json
+	 *   - csv
+	 *   - yaml
+	 *   - count
+	 * ---
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     # List all un-autoloaded options
+	 *     $ wp option autoload list no
+	 *     +-------------------+
+	 *     | option_name       |
+	 *     +-------------------+
+	 *     | moderation_keys   |
+	 *     | recently_edited   |
+	 *     | blacklist_keys    |
+	 *     | uninstall_plugins |
+	 *     +-------------------+
+	 *
+	 * @subcommand list
+	 */
+	public function list_( $args, $assoc_args ) {
+
+		list( $yn ) = $args;
+		$yn = $this->validate_yn( $yn, 'yes' );
+
+		// option list uses on/off ¯\_(ツ)_/¯
+		$yn = 'yes' === $yn ? 'on' : 'off';
+
+		WP_CLI::run_command(
+			[ 'option', 'list' ],
+			[
+				'autoload' => $yn,
+				'fields' => 'option_name',
+				'format' => $assoc_args['format'],
+			],
```

Comment:
This comma is valid, but I hate it :)

Reply:
Well, it's only valid on PHP 7.3+ So I'm not keeping it upstream.

Reply:
SVN pre-commit hook didn't like it either. It's been removed in master.